File: redome-workup/src/main/java/br/org/cancer/redome/workup/model/AvaliacaoPedidoColeta.java
package br.org.cancer.redome.workup.model;

import java.io.Serializable;
import java.time.LocalDateTime;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Builder.Default;
import lombok.Data;
import lombok.NoArgsConstructor;


/**
 * Classe de persistência para avaliação do pedido de coleta.
 * Nesta entidade será gravada a avaliação que o médico do REDOME fará
 * em caso de pedido de coleta do doador.
 * @author Filipe Paes
 * 
 * @author ergomes
 *
 */
@Entity
@Table(name="AVALIACAO_PEDIDO_COLETA")
@SequenceGenerator(name = "SQ_AVPC_ID", sequenceName = "SQ_AVPC_ID", allocationSize = 1)
@Data
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
public class AvaliacaoPedidoColeta implements Serializable {
	
	private static final long serialVersionUID = -7526609614339948073L;

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "SQ_AVPC_ID")
	@Column(name="AVPC_ID")
	private Long id;

	@Column(name="AVPC_DT_CRIACAO")
	@Default
	private LocalDateTime dataCriacao = LocalDateTime.now();

	@Column(name="AVPC_IN_PEDIDO_PROSSEGUE")
	private Boolean pedidoProssegue;

	@Column(name="AVPC_TX_JUSTIFICATIVA")
	private String justificativa;

	@ManyToOne
	@JoinColumn(name="AVRW_ID")
	private AvaliacaoResultadoWorkup avaliacaoResultadoWorkup;
	
	@Column(name="USUA_ID")
	private Long usuario;

	
		
}

File: redome-front/src/app/doador/consulta/coleta/avaliacao/avaliacao.pedido.coleta.ts
import { UsuarioLogado } from '../../../../shared/dominio/usuario.logado';
import { Solicitacao } from '../../../solicitacao/solicitacao';
import { BaseEntidade } from '../../../../shared/base.entidade';


/**
 * Classe de representação de Avaliação de Pedido de Coleta.
 * @author Filipe Paes
 */
export class AvaliacaoPedidoColeta extends BaseEntidade {
    public jsonToEntity(res: any) {
        throw new Error("Method not implemented.");
    }

    private _id: Number;

    private _dataAvaliacao: Date;

    private _dataCriacao: Date;

    private _pedidoProssegue: Boolean;

    private _justificativa: String;

    private _solicitacao: Solicitacao;

    private _usuario: UsuarioLogado;


    public get id(): Number {
        return this._id;
    }

    public set id(value: Number) {
        this._id = value;
    }


    public get dataAvaliacao(): Date {
        return this._dataAvaliacao;
    }

    public set dataAvaliacao(value: Date) {
        this._dataAvaliacao = value;
    }


    public get dataCriacao(): Date {
        return this._dataCriacao;
    }

    public set dataCriacao(value: Date) {
        this._dataCriacao = value;
    }


    public get pedidoProssegue(): Boolean {
        return this._pedidoProssegue;
    }

    public set pedidoProssegue(value: Boolean) {
        this._pedidoProssegue = value;
    }


    public get justificativa(): String {
        return this._justificativa;
    }

    public set justificativa(value: String) {
        this._justificativa = value;
    }


    public get solicitacao(): Solicitacao {
        return this._solicitacao;
    }

    public set solicitacao(value: Solicitacao) {
        this._solicitacao = value;
    }


    public get usuario(): UsuarioLogado {
        return this._usuario;
    }

    public set usuario(value: UsuarioLogado) {
        this._usuario = value;
    }

}